docs/kb/semgrep-code/collect-cli-logs.md

```diff
@@ -0,0 +1,77 @@
+---
+tags:
+  - Semgrep CLI
+  - Logs
+description: How to collect logs when running Semgrep on the command line.
+---
+
+# How to collect logs when running Semgrep in CLI
+
+When troubleshooting Semgrep scans on the command line (CLI), collecting and sharing logs can be extremely helpful. By default, Semgrep prints findings from a scan to `stdout`, and other messages, including scan details and progress, to `stderr`. For troubleshooting, it's most often useful to provide both.
+
+To collect all relevant logs for a scan, follow these instructions. All log output options apply to both `semgrep scan` and `semgrep ci`. All examples use `semgrep ci` for simplicity, and name the output file as `semgrep.log`.
+
+## Capturing full log output
+
+To store the entire Semgrep log for a scan, including the findings:
+
+```
+semgrep ci &> semgrep.log
+```
+
+## Separating findings and other logs
+
+Sometimes it's helpful to separate findings from other scan logs. Using the following commands separates the two and allows for independent review of findings and scan behavior.
+
+Capture only findings in a file, print other logs to the terminal:
+
+```
+semgrep ci > semgrep.log
+```
+
+Capture findings and other logs independently, each within files:
+
+```
+semgrep ci -o findings.txt 2> semgrep.log
+semgrep ci > findings.txt 2> semgrep.log
+```
+
+Capture other logs in a file, print findings to the terminal:
+
+```
+semgrep ci 2> semgrep.log
+```
+
+### Formatting findings
+
+Semgrep can output findings in a variety of formats. By default, the findings are formatted as readable text in the terminal, but they can also be output in other formats such as JSON or SARIF. For example:
+
+```
+semgrep ci --json -o findings.json 2> semgrep.log
+```
+
+outputs findings as JSON and saves the scan log to `semgrep.log`.
+
+In addition to findings formats, there are options to add details of the data flow (`--dataflow-traces`) or explanations of rule matching (`--matching-explanations`). These are less frequently used in overall scan troubleshooting, but can be helpful for understanding findings.
+
+## Logging verbosity options
+
+Semgrep has three commonly used log levels.
```

Comment:
```suggestion
Semgrep has three logging levels.
```

Reply:
I struggle with simplifying things to this degree because it's not strictly correct. Since I also mention quiet mode (which is a kind of log level) I think I'm going to pass this one up.